### PR TITLE
feat(engine): support bpmn converging inclusive gateway

### LINKED ISSUE #10031: Support BPMN Converging Inclusive Gateway
**Is your feature request related to a problem? Please describe.**
The BPMN symbol for the inclusive gateway is supported with
- #9747 

However, only the diverging behavior was added. Users may want to use the converging behavior as well.

**Describe the solution you'd like**
- remove the validation that restricts the number of incoming sequence flows to an inclusive gateway to max 1.
- only activate the inclusive gateway when (see discussion https://github.com/camunda/zeebe/pull/9747#discussion_r925770530):
  - there are no active children of its flow scope instance (we can check this already),
  - or if all incoming sequence flows were taken at least once (we can check this already),
  - or if no path can be found from any of the active children to the inclusive gateway, excluding paths that end on an already taken incoming sequence flow of the gateway. (but we can't check this yet).

**Describe alternatives you've considered**
Users can use a combination of parallel and exclusive gateways to build similar joining logic.

**Additional context**
Spec: https://www.omg.org/spec/BPMN/2.0.2/PDF

Product-Hub:
- https://github.com/camunda/product-hub/issues/364

_EDIT: Update the solution about active children based on the BPMN spec_

### PR DESCRIPTION
**Pull request by lzgabel**
## Description
This PR adds support for converging/joining inclusive gateway.

## Related issues
closes #10031

## Definition of Done
Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by lzgabel** [comment]
Hi @camunda/zeebe-process-automation. Please take a moment to review this feature. 🙇

**Comment by koevskinikola** [comment]
Hey @lzgabel,

Thank you for submitting this contribution! I'll review the code this week and provide you with feedback.

Cheers,
Nikola

File: engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/gateway/InclusiveGatewayTest.java [hunk 1]
```diff
@@ -676,6 +677,246 @@ public void shouldEvaluateConditionWithoutDefaultFlow() {
...
+    assertThat(events)
+        .extracting(e -> e.getValue().getElementId(), Record::getIntent)
+        .containsSubsequence(
+            tuple("joinFlow1", ProcessInstanceIntent.SEQUENCE_FLOW_TAKEN),
+            tuple("joinFlow2", ProcessInstanceIntent.SEQUENCE_FLOW_TAKEN))
+        .containsOnlyOnce(tuple("join", ProcessInstanceIntent.ELEMENT_ACTIVATING));
+  }
+
+  @Test
+  public void shouldMergeAndSplitInOneGateway() {
```
Comment (koevskinikola) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/gateway/InclusiveGatewayTest.java:884]: :+1: Great test coverage on the new converging inclusive gateway behavior!

:x: However, I think we should add a test case when one gateway splits and merges, but only one, or a subset of the incoming sequence flows are activated.

**Review by koevskinikola — CHANGES_REQUESTED** [review]
The changes look great @lzgabel. :rocket: 

I just have some small feedback for the test coverage. You can see it in the comment below.

**Comment by lzgabel** [comment]
Hi @koevskinikola . Please check this out. :heart:

**Comment by lzgabel** [comment]
🤔 I received this emai. What's wrong? Am I missing something?

86F8D1D6-8078-467A-8045-0549873130CC

**Comment by koevskinikola** [comment]
@lzgabel the email is a bit strange, I haven't triggered any benchmark workflows. I'll check with the team if someone changed anything regarding the PR workflows.

Thank you for reporting this!

**Review by koevskinikola — APPROVED** [review]
@lzgabel seems like the PR benchmark CI job was triggered by accident and it failed due to some flaky tests.

The PR looks good now, thank you for providing support for the converging inclusive gateway! :rocket: 

Cheers!

bors merge

**Pull request merged**